### ISSUE TITLE
PHP errors when changing the editor on an existing text format

### ISSUE THREAD

**Issue by hosef** [backdrop-contrib/ckeditor5#38]
I found 2 errors that show up after changing the editor on an existing text format from ckeditor to ckeditor5.

When trying to edit a text format after changing the editor:
```
TypeError: Illegal offset type in isset or empty in template_preprocess_ckeditor5_settings_toolbar() (line 35 of /srv/http/projects/backdrop/ckeditor5/ckeditor5.theme.inc).
```

When viewing any content that has a long text field:
```
TypeError: Illegal offset type in isset or empty in ckeditor5_get_config() (line 686 of /srv/http/projects/backdrop/ckeditor5/ckeditor5.module).
```

**Comment by indigoxela** [comment]
Same error here.

**Comment by indigoxela** [comment]
Currently just using CKE5 on PHP 8.1 more or less breaks everything. I think, this should be the first thing to fix.
Then switching from 4 to 5.

**Comment by indigoxela** [comment]
I played with it for a while now, and want to share my findings:

Switching between 4 and 5 will require changes in core.

1. CKEditor 4 happily initializes with the wrong path (we could fix that with a global js variable for CKEDITOR_BASEPATH, but that has to be dynamic (basePath can be different for installations). This is necessary for both, switching on node form and switching on admin form.
2. The structure difference in editor settings (string vs. array for buttons) needs to be taken care of, when switching on the admin form, so both functions, template_preprocess_ckeditor_settings_toolbar() and template_preprocess_ckeditor5_settings_toolbar() can handle a toggle.

**Comment by indigoxela** [comment]
Here's the core issue re switching between two editors: https://github.com/backdrop/backdrop-issues/issues/6203

The console.log nagging when both modules are on and load on the same page is a bit tricky, too.
When CKE4 initializes, it (reliably :zany_face:) grabs the wrong basePath, as the CKE5 js file just has the same name (ckeditor.js). The most reliable solution is to change the name of the file in lib/ckeditor5/build/ to ckeditor5.js - but that requires an additional step for library updates.
I also played with setting the global var CKEDITOR_BASEPATH before the file gets loaded. And I also played with setting the CKE5 lib weight to something higher/lower. No final conclusion, yet, but feedback is welcome.
